schism/compiler.ss

```diff
@@ -237,13 +250,13 @@
               (if (string-equal? s (car ls))
                   (%set-tag ls ,(symbol-tag))
                   (%find-symbol-by-name s (cdr ls)))))
-       (define (gensym t) ;; Creates a brand new symbol that cannot be reused
-         (let ((x (cons #f '())))
+       (define (gensym str) ;; Creates a brand new symbol that cannot be reused
+         (let ((x (cons (string->list str) '())))
```

Comment:
Could this just be `str` instead of `(string->list str)`?

Reply:
It certainly could, but "display" should somehow be able to differentiate between gensyms and interned symbols, and write gensyms as `#<gensym ...>`.  The string->list is only for that purpose.

If you are thinking of #55, then this could instead be a string->vector or something.

Reply:
Ah, sounds good to me. At some point, we'll probably have the symbol table keep track of gensyms too so we could write them as `#<gensym foo.N>` or something like that. At that point, we can just as, "is this a gensym?"

Reply:
Oh, duh, you actually are printing gensyms out that way already; that's the point of the CL :)